### PR TITLE
fix choose cohorts close or cancel updating feature cohorts selection in flyout

### PR DESCRIPTION
**Pull request by imatiach-msft**
## Description

Fixing bug from bug bash discovered by @vinuthakaranth 
The "Choose cohorts" flyout was saving the user's selection even when they clicked cancel or close buttons.
This issue was found during text/vision dashboard bugbash but it looks like a general issue that is in the tabular dashboard as well.

Video of functionality prior to fix:

https://github.com/microsoft/responsible-ai-toolbox/assets/24683184/5afe7832-69fe-43b6-aa03-da0ed53e7bc3

Video of functionality after fix:

https://github.com/microsoft/responsible-ai-toolbox/assets/24683184/f0f9b577-dedd-4ad7-ad54-71f0270a79d2

## Checklist

- [ ] I have added screenshots above for all UI changes.
- [ ] I have added e2e tests for all UI changes.
- [ ] Documentation was updated if it was needed.

### CODE REVIEW

File: libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/ChartConfigurationFlyoutProps.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { ErrorCohort } from "@responsible-ai/core-ui";
+
+export interface IChartConfigurationFlyoutProps {
```
Comment (vinuthakaranth) [inline libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/ChartConfigurationFlyoutProps.tsx:6]: nit: both props and state interface can be in single file

Comment (vinuthakaranth) [reply libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/ChartConfigurationFlyoutProps.tsx:6]: or can be moved to utils file too

File: libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/ChartConfigurationFlyoutUtils.tsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+import { IDropdownOption } from "@fluentui/react";
+import { ErrorCohort } from "@responsible-ai/core-ui";
+
+import { IChartConfigurationFlyoutProps } from "./ChartConfigurationFlyoutProps";
+
+export const selectAllOptionKey = "selectAll";
+
+export function getInitialNewlySelectedDatasetCohorts(
+  props: IChartConfigurationFlyoutProps
```
Comment (vinuthakaranth) [inline libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/ChartConfigurationFlyoutUtils.tsx:12]: can we just pass the props that are needed to this function and the following function? this avoids the overhead of passing all unwanted props.

**Review by vinuthakaranth — APPROVED** [review]

File: libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/ChartConfigurationFlyoutProps.tsx (hunk 1 above)
Comment (imatiach-msft) [reply libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/ChartConfigurationFlyoutProps.tsx:6]: we have 24 files that are props only:
https://github.com/search?q=repo%3Amicrosoft%2Fresponsible-ai-toolbox+Props.ts&type=code
and we have 12 files that are state only:
https://github.com/search?q=repo%3Amicrosoft%2Fresponsible-ai-toolbox+State.ts&type=code
I hope it's ok to keep to this convention?  I just wanted to bring the line count down and this was the only thing I could think of.  Seems it needs deeper refactoring to bring it below the suggested limit of 300 lines.

File: libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/ChartConfigurationFlyoutUtils.tsx (hunk 2 above)
Comment (imatiach-msft) [reply libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/ChartConfigurationFlyoutUtils.tsx]: good point, done!

**Pull request merged**